### DISCUSSION TITLE
Static Site Hosting

### DISCUSSION THREAD

**Discussion by rushsteve1** [supabase/supabase#991] [Feature Requests]
### Update: this is currently not on the roadmap

---

The ability to host frontend websites (SPAs, Jamstack) on Supabase would make it an ideal alternative to Netlify and Vercel, especially once functions are added. Custom domain support probably be necessary, though likely could just be a CNAME.

This was my first though on reading the Supabase Storage announcement, and it seems like a logical next step to me.

A build system similar to Netlify's CI/CD would also be an amazing addition, though is obviously much more complex than just hosting the pre-built files.

I hadn't seen anyone mention this idea anywhere, so I thought I would bring it up.

**Comment by boian-ivanov** [comment]
Are there any updates on this topic? Seems like it's a no brainer to have a Storage bucket that would host your build files. It just needs to be exposed, so you can point your DNS to it. It would also be great to have a CD pipeline that can do all of that automatically (like Netlify, Vercel, Amplify, etc), but I'd say that's a good to have.

**Comment by ghost** [comment]
Would love to see this feature!

**Comment by AminDhouib** [comment]
I am all for it too! This would make Supabase feel very complete!

**Comment by AndreBooysen** [comment]
Yes please, that's the only piece of my project that is not on Supabase and it is functionality that's available on aws buckets directly. It just needs to be exposed on Supabase

**Comment by zaceno** [comment]
Chiming in! I too would like to see ability to use supabase as a web-server of static assets. Firebase can do it, and I'm trying to move away from firebase to supabase.

**Comment by rustygreen** [comment]
Any update on this?

**Comment by joaoantunes-mobiweb** [comment]
I was looking for the same, since I was used to do it with firebase. This would be a great feature.

**Reply by probert94** [reply]
Supabase calls itself "an open source Firebase alternative" and it is probably more then just an alternative as it offers a full SQL database.
However hosting of static websites is still missing and would be a very good addition, so that I don't have to use another service just for that.

**Reply by AndreBooysen** [reply]
@probert94, I used to feel the same until I started working with Vercel+Supabase. I totally support the direction they took with the Vercel integration, it just works great.

**Reply by probert94** [reply]
@AndreBooysen I just started a project so I did no deployement yet, but I'll take a look at it!
Thank you

**Comment by joaomelo** [comment]
Maybe you could partner with one or more of those providers and advance an integrated hosting solution. This kind of full stack makes a lot of difference in the lives of small teams. The transparent CI/CD makes the most difference, but it even opens the possibility of some magic, like inserting keys directly by the hosting server.

**Reply by MildTomato** [reply]
Hey @joaomelo - We've been putting more effort into our Vercel integration which will also support our upcoming Branching support as well. This includes providing keys to Vercel for prod and preview deployments. 

Any other hosting providers we should be thinking about?

**Reply by joaomelo** [reply]
Hi @MildTomato. Thank you for getting back to me. 

It's nice to know you folks are investing in this front. Cloudflare and Netfly may also be good options. 

I use Firebase hosting mainly because it has everything there (hosting, db, auth, object, etc.). Still, I wonder if it is a sound business strategy for Supabase to invest in this integration.

I wish the best for you.

**Reply by zaceno** [reply]
@MildTomato From my perspective, the strength of supabase is that it is an open-source, cloud-agnostic all-in-one set up. (almost all-in-one - It's just missing the serving of static web-assets)

In my dream world I would like to be able to self-host my entire system - including a web-server - locally or on a VM provider of my choice, and manage it all from the project-admin interface (including things like routing and access-protection of routes). Integrating with popular hosting providers is  fine, but giving me something I can control and easily move around to wherever, in a unified package, would be 💯

**Comment by alexpineda** [comment]
I'm presently facing this issue and would just like everything in supabase instead of using cloudflare pages.

**Comment by ghost** [comment]
I would like to be able to easily put my classic RESTish API server on the same network as my managed Supabase database. If I deploy it to Google CloudRun, Linode, DigitalOcean App Platform, etc. they won't even be located in the same geographical location as the Supabase DB server's AWS region.

**Comment by stephenhandley** [comment]
Just saw this in the Launch Week Announcement
https://supabase.com/blog/postgrest-12#media-type-handlers

**Comment by shahil-kv** [comment]
can the core team please discuss about the hosting in supabase if there is hosting also then i will be a completely supabase user otherwise i need to rely on otherservices for hosting please make this request possible waiting for this feature very long 
thank you

**Comment by hogypitersan** [comment]
i guess that in 2024 this feature is still out of our dream.
Really need this feature right now, so the Supabase can be one stop service that all we need.

**Comment by madalinignisca** [comment]
This feature request to match a missing important component of Firebase is probably intentionally ignored, as it would compete with parties using and supporting commercial supabase.

Once you have it and self-host it, what's the reason to use any other service to serve our wonderful Next and Nuxt apps?

For example, Vercel or others do way more stuff to serve a Nextjs app.

However, running it with some modifications static + functions in Firebase and on Cloudflare has been possible for a reasonable amount of time now.

Adding a static site like on Firebase should be enough for people to come up with a solution to use it with functions to get the full stack all on the supabase.

**Reply by oscarryz** [reply]
I wonder if now that Storage is public, this can be used to host apps

https://supabase.com/docs/guides/storage/serving/downloads

**Reply by robertoandrade** [reply]
Looks like it's possible but you'd need to either have a reverse proxy in front of it to rewrite the urls to `/v1/object/public/<bucket>/<file.ext>`. I've been using CloudFlare's Workers offering for that purpose (which is free) but at that point if you're using something else to front this you might as well use their S3-like storage bucket to sync from git automatically and front it. Indeed would be something very useful for Supabase to implement given the seemly matching offering from Firebase.

**Comment by ianzone** [comment]


**Reply by IanMayo** [reply]
That's just spam, IMHO.

**Reply by ianzone** [reply]
??? I'm not affiliated

**Reply by IanMayo** [reply]
I know, but it referred to a separate, apparently unrelated app. The comment doesn't help move this issue forward.
